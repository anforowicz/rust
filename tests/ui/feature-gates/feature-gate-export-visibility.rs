// This test verfies that `#[export_visibility = ...]` cannot be used without
// opting into the corresponding unstable feature via
// `#![feature(export_visibility)]`.
#[export_visibility = "hidden"]
// `#[export_name = ...]` is present to avoid hitting the following error:
// export visibility will be ignored without `export_name`, `no_mangle`, or similar attribute
#[unsafe(export_name = "exported_static")]
pub static TESTED_STATIC: [u8; 6] = *b"foobar";

fn main() {}
