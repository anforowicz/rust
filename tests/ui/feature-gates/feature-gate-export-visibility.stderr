error[E0658]: `#[export_visibility = ...]` is currently unstable
  --> $DIR/feature-gate-export-visibility.rs:4:1
   |
LL | #[export_visibility = "hidden"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #123456 <https://github.com/rust-lang/rust/issues/123456> for more information
   = help: add `#![feature(export_visibility)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0658`.
